subsys/bluetooth/shell/flash.c

```diff
@@ -0,0 +1,227 @@
+/** @file
+ * @brief Bluetooth Controller and flash co-operation
+ *
+ */
+
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <shell/shell.h>
+#include <misc/printk.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include "flash.h"
+#include <soc.h>
+
+#define FLASH_SHELL_MODULE "flash"
+#define BUF_ARRAY_CNT 16
+#define TEST_ARR_SIZE 0x1000
+
+u8_t test_arr[TEST_ARR_SIZE];
+
+static int cmd_erase(int argc, char *argv[])
+{
+	struct device *flash_dev;
+	u32_t page_addr;
+	int result;
+	u32_t size;
+
+	flash_dev = device_get_binding(CONFIG_SOC_FLASH_NRF5_DEV_NAME);
+
+	if (!flash_dev) {
```

Comment:
Whenever you have an assignment to a variable and then a test for the variable, there should be no empty line in between (to indicate that the two are related to each other). I was going to let this go since you've had quite many revisions already and this is not super critical, however there were still several other style issues I spotted as well.